src/commands.ts

```diff
@@ -113,3 +136,46 @@ export async function exportPluginFromContextMenu(uri: vscode.Uri): Promise<void
     // Store selected extensions so we don't have to ask again
     fileutils.storeFileExtensionConfig(path.dirname(scriptPath), extensions);
 }
+
+async function createDataPluginFromSampleFile(dataPlugin: DataPlugin): Promise<DataPlugin | null> {
+    const openDialogOptions = {
+        canSelectFiles: true,
+        canSelectFolders: false,
+        canSelectMany: false,
+        title: 'Choose your sample data file'
+    };
+
+    const selectedFiles = await vscode.window.showOpenDialog({ ...openDialogOptions });
+    if (selectedFiles) {
+        const sampleFile = selectedFiles[0];
+        // copy sample file to workspace
+        try {
+            const sampleFileName = path.basename(sampleFile.fsPath);
+            await fs.copy(sampleFile.fsPath, path.join(dataPlugin.folderPath, sampleFileName));
+        } catch (e) {
+            void vscode.window.showErrorMessage('Could not copy sample file to workspace.');
```

Comment:
I would usually prefer to include the exception message in the error message. Otherwise the user has no clue, why it happend or what he can do to fix it.

Reply:
yes right! 👍